src/mono/mono/metadata/object.c

```diff
@@ -1576,7 +1578,7 @@ build_imt_slots (MonoClass *klass, MonoVTable *vt, gpointer* imt, GSList *extra_
 			for (method_slot_in_interface = 0; method_slot_in_interface < mcount; method_slot_in_interface++) {
 				MonoMethod *method = mono_class_get_method_by_index (iface, method_slot_in_interface);
 
-				if (method->is_generic)
+				if (method->is_generic && m_method_is_virtual(method))
 					has_generic_virtual = TRUE;
```

Comment:
Pretty sure this is dead code, but I wanted to highlight the potential problem.

Reply:
yea, I ended up deleting this in #93349 - it was only used for COM interop